nameresolution/sqlite/sqlite.go

```diff
@@ -0,0 +1,317 @@
+			`DELETE FROM %s WHERE unixepoch(CURRENT_TIMESTAMP) - last_update < %d`,
+			s.metadata.TableName,
+			int(s.metadata.UpdateInterval.Seconds()),
+		),
+		CleanupInterval: s.metadata.CleanupInterval,
+		DB:              internalsql.AdaptDatabaseSQLConn(s.db),
+	})
+	return err
+}
+
+// Registers the host
+func (s *resolver) registerHost(ctx context.Context) error {
+	// Get the registration ID
+	u, err := uuid.NewRandom()
+	if err != nil {
+		return fmt.Errorf("failed to generate registration ID: %w", err)
+	}
+	s.registrationID = u.String()
+
+	queryCtx, queryCancel := context.WithTimeout(ctx, s.metadata.Timeout)
+	defer queryCancel()
+
+	// There's a unique index on address
+	// We use REPLACE to take over any previous registration for that address
+	// TODO: Add support for namespacing. See https://github.com/dapr/components-contrib/issues/3179
+	_, err = s.db.ExecContext(queryCtx,
+		fmt.Sprintf("REPLACE INTO %s (registration_id, address, app_id, namespace, last_update) VALUES (?, ?, ?, ?, unixepoch(CURRENT_TIMESTAMP))", s.metadata.TableName),
+		s.registrationID, s.metadata.GetAddress(), s.metadata.appID, "",
+	)
+	if err != nil {
+		return fmt.Errorf("failed to register host: %w", err)
+	}
+
+	return nil
+}
+
+// In backgrounds, periodically renews the host's registration
+// Should be invoked in a background goroutine
+func (s *resolver) renewRegistration() {
+	defer s.wg.Done()
+
+	addr := s.metadata.GetAddress()
+
+	d := s.metadata.UpdateInterval - s.metadata.Timeout
+	s.logger.Debugf("Started renewing host registration in background with interval %v", s.metadata.UpdateInterval)
+	t := time.NewTicker(d)
+	defer t.Stop()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for {
+		select {
+		case <-s.closeCh:
+			// Component is closing
+			s.logger.Debug("Stopped renewing host registration: component is closing")
+			return
+
+		case <-t.C:
+			// Renew on the ticker
+			go func() {
+				err := s.doRenewRegistration(ctx, addr)
+				if err != nil {
+					// Log errors
+					s.logger.Errorf("Failed to update host registration: %v", err)
+
+					if errors.Is(err, errRegistrationLost) {
+						// This means that our registration has been taken over by another host
+						// It should never happen unless there's something really bad going on
+						// Panicking here to force a restart of Dapr
+						s.logger.Fatalf("Host registration lost")
+					}
+				}
+			}()
+		}
+	}
+}
+
+func (s *resolver) doRenewRegistration(ctx context.Context, addr string) error {
+	queryCtx, queryCancel := context.WithTimeout(context.Background(), s.metadata.Timeout)
+	defer queryCancel()
+
+	res, err := s.db.ExecContext(queryCtx,
+		fmt.Sprintf("UPDATE %s SET last_update = unixepoch(CURRENT_TIMESTAMP) WHERE registration_id = ? AND address = ?", s.metadata.TableName),
+		s.registrationID, addr,
+	)
+	if err != nil {
+		return fmt.Errorf("database error: %w", err)
+	}
+
+	n, _ := res.RowsAffected()
```

Comment:
Check error, and return error message if it not nil.

Reply:
I think that's redundant. If `err != nil`, then `n == 0` as a consequence. It will error anyways